app/controllers/posts_controller.rb

```diff
@@ -27,4 +28,21 @@ def create
       render :new
     end
   end
+
+  def destroy
+    @post = Post.find_by(author_id: params[:user_id], id: params[:post_id])
+    return render file: "#{Rails.root}/public/404.html", status: :not_found, layout: false if @post.nil?
+
+    if @post.destroy
+      redirect_to user_posts_path(current_user)
+    else
+      redirect_to user_post_path(current_user, @post)
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:title, :text)
+  end
```

Comment:
- [ ] Great job on adding a destroy action in the post controller, however in the requirement we should also add this action for comments. A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Kindly add this action.

<img width="763" alt="image" src="https://user-images.githubusercontent.com/59416136/224133611-17dac832-ebc2-4e17-bb08-a44ae4400f1e.png">